src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php

```diff
@@ -53,47 +89,328 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
+                        $platformTopology->getAddress()
+                    )
+                );
+            }
+            if (
+                null === $foundPlatformInformation->getApiUsername()
+                || null === $foundPlatformInformation->getApiCredentials()
+            ) {
+                throw new PlatformTopologyException(
+                    sprintf(
+                        _("Central's credentials are missing on: '%s'@'%s'. Please check the Remote Access form."),
+                        $platformTopology->getName(),
+                        $platformTopology->getAddress()
+                    )
+                );
+            }
+            if (null === $foundPlatformInformation->getApiScheme()) {
+                throw new PlatformTopologyException(
+                    sprintf(
+                        _("Central's protocol scheme is missing on: '%s'@'%s'. Please check the Remote Access form."),
                         $platformTopology->getName(),
                         $platformTopology->getAddress()
                     )
                 );
             }
-            $platformTopology->setServerId($foundCentralInNagiosTable->getId());
+            if (null === $foundPlatformInformation->getApiPort()) {
+                throw new PlatformTopologyException(
+                    sprintf(
+                        _("Central's protocol port is missing on: '%s'@'%s'. Please check the Remote Access form."),
+                        $platformTopology->getName(),
+                        $platformTopology->getAddress()
+                    )
+                );
+            }
+            if (null === $foundPlatformInformation->getApiPath()) {
+                throw new PlatformTopologyException(
+                    sprintf(
+                        _("Central's path is missing on: '%s'@'%s'. Please check the Remote Access form."),
+                        $platformTopology->getName(),
+                        $platformTopology->getAddress()
+                    )
+                );
+            }
+
+            /**
+             * Getting platform's proxy data
+             * @var Proxy|null $proxy
+             */
+            $proxy = $this->proxy->getProxy();
+
+            /**
+             * Call the API on the n-1 server to register it too
+             */
+            try {
+                // Central's API endpoints base path
+                $baseApiEndpoint = $foundPlatformInformation->getApiScheme() . '://' .
+                    $foundPlatformInformation->getAuthorizedMaster() . ':' .
+                    $foundPlatformInformation->getApiPort() . '/' .
+                    $foundPlatformInformation->getApiPath() . '/api/v2.0/';
+
+                // Enable specific options
+                $optionPayload = [];
+                // Enable proxy
+                if (null !== $proxy && !empty($proxy->__toString())) {
+                    $optionPayload['proxy'] = $proxy->__toString();
```

Comment:
```suggestion
                if (null !== $proxy && !empty((string) $proxy)) {
                    $optionPayload['proxy'] = (string) $proxy;
```
The magical Method __toString(); will automatically format your class instancy when you need to use it in the context of a string (here casting string will trigger __toString() )